src/components/Select2.js

```diff
@@ -46,6 +51,22 @@ export default class Select2 extends Component {
     this.props.events.forEach(event => {
       this.el.on(event[0], this.props[event[1]]);
     });
+    const { defaultValue, value } = this.props;
+    if (!defaultValue && this.el.val() !== value) {
```

Comment:
You should remove this.el.val() !== value as it is the same check as on line 69.